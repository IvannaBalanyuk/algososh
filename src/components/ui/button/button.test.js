import React from "react";
import renderer from "react-test-renderer";

import { Button } from "./button";
import { fireEvent, render, screen } from "@testing-library/react";

describe("Button - корректность отрисовки", () => {
  let buttonElement;

  afterEach(() => {
    expect(buttonElement).toMatchSnapshot();
  });

  test("Button с текстом рендерится без ошибок", () => {
    buttonElement = renderer.create(<Button linkedList="small" text="Нажми на меня" />).toJSON();
  });

  test("Button без текста рендерится без ошибок", () => {
    buttonElement = renderer.create(<Button linkedList="small" />).toJSON();
  });

  test("Button в состоянии disabled рендерится без ошибок", () => {
    buttonElement = renderer.create(<Button linkedList="small" disabled />).toJSON();
  });

  test("Button с индикацией загрузки рендерится без ошибок", () => {
    buttonElement = renderer.create(<Button linkedList="small" isLoader={true} />).toJSON();
  });
});

describe("Button - корректность вызова коллбэка", () => {
  test("При нажатии на кнопку коллбэк вызывается корректно", () => {
    const callback = jest.fn();

    render(<Button text='Нажми на меня' onClick={callback} />);

    const button = screen.getByText('Нажми на меня');

    expect(callback).toHaveBeenCalledTimes(0);
    fireEvent.click(button);
    expect(callback).toHaveBeenCalledTimes(1);
  });
});
