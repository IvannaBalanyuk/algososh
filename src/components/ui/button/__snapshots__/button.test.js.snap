// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button - корректность отрисовки Button без текста рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button small false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button - корректность отрисовки Button в состоянии disabled рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button small false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button - корректность отрисовки Button с индикацией загрузки рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button small loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button - корректность отрисовки Button с текстом рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button small false "
  type="button"
>
  <p
    className="text undefined"
  >
    Нажми на меня
  </p>
</button>
`;
